Docker Build Args not working

I'm trying to pass build arguments using an external Dockerfile, but fail to get it working

global properties:

`<properties>
        <java.version>1.8</java.version>
        <fabric8.maven.plugin.version>4.2.0</fabric8.maven.plugin.version>
        <fabric8.mode>openshift</fabric8.mode>
        <fabric8.build.strategy>docker</fabric8.build.strategy>
    </properties>`

plugin config:

`                <artifactId>fabric8-maven-plugin</artifactId>
                <version>4.20</version>
                <configuration>
                    <images>
                        <image>
                            ...
                            <build>
                                <dockerFileDir>${project.basedir}</dockerFileDir>
                                <assembly>
                                    <name>target/</name>
                                    <descriptorRef>artifact</descriptorRef>
                                </assembly>
                                <args>
                                    <myVAR>TEST</myVAR>
                                </args>
                            </build>
                        </image>
                    </images>
                </configuration>`

Dockerfile snippet:

`FROM openjdk:11
ARG myVAR
VOLUME /tmp`

But myVAR is always empty. 

* f-m-p version : 4.20
* Maven version 3.6
* Kubernetes / OpenShift setup and version : 1.11.0/3.11.0

@CurryWurry : Hi, I'm not sure if I tested it properly but I faced a different error when building:
```
[INFO] --- fabric8-maven-plugin:4.3-SNAPSHOT:build (default-cli) @ fabric8-maven-sample-external-resources ---                                                
[INFO] F8: Running in OpenShift mode                                                                                                                          
[INFO] F8: Using OpenShift build with strategy Docker
[INFO] Copying files to /home/rohaan/work/repos/fmp-sample-external-resources/target/docker/external-resources/build/target
[WARNING] F8: Dockerfile /home/rohaan/work/repos/fmp-sample-external-resources/Dockerfile does not contain an ADD or COPY directive to include assembly create
d at target/. Ignoring assembly.
[INFO] Building tar: /home/rohaan/work/repos/fmp-sample-external-resources/target/docker/external-resources/tmp/docker-build.tar
[INFO] F8: [external-resources:latest]: Created docker source tar /home/rohaan/work/repos/fmp-sample-external-resources/target/docker/external-resources/tmp/d
ocker-build.tar
[INFO] F8: Creating BuildServiceConfig external-resources-s2i for Docker build
[INFO] F8: Creating ImageStream external-resources
[INFO] F8: Starting Build external-resources-s2i
[INFO] F8: Waiting for build external-resources-s2i-1 to complete...
[INFO] F8: 
[INFO] F8: Pulling image openjdk:11 ...
[INFO] F8: Pulled 0/8 layers, 10% complete
[INFO] F8: Pulled 1/8 layers, 18% complete
[INFO] F8: Pulled 2/8 layers, 29% complete
[INFO] F8: Pulled 3/8 layers, 44% complete
[INFO] F8: Pulled 4/8 layers, 65% complete
[INFO] F8: Pulled 5/8 layers, 66% complete
[INFO] F8: Pulled 6/8 layers, 77% complete
[INFO] F8: Pulled 7/8 layers, 88% complete
[INFO] F8: Pulled 7/8 layers, 88% complete
[ERROR] F8: Failed to execute the build: io.fabric8.maven.core.service.Fabric8ServiceException: Unable to build the image using the OpenShift build service
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  01:34 min
[INFO] Finished at: 2019-10-01T19:59:24+05:30
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal io.fabric8:fabric8-maven-plugin:4.3-SNAPSHOT:build (default-cli) on project fabric8-maven-sample-external-resources: Failed to 
execute the build: io.fabric8.maven.core.service.Fabric8ServiceException: Unable to build the image using the OpenShift build service -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
```

I tested using this project: https://github.com/r0haaaan/fmp-openshift-docker-buildmode

Could you please confirm if the reproducing sample has correct configuration?

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -11,6 +11,8 @@ We use semantic versioning in some slight variation until our feature set has st
 After this we will switch probably to real [Semantic Versioning 2.0.0](http://semver.org/)
 
 ### 4.3-SNAPSHOT
+* Fix #1725: Update jackson to 2.10.0
+* Fix #1710: Docker build args not working
 
 ### 4.3.0 (04-10-2019)
 * Updated custom-enricher sample

--- a/core/src/main/java/io/fabric8/maven/core/service/openshift/OpenshiftBuildService.java
+++ b/core/src/main/java/io/fabric8/maven/core/service/openshift/OpenshiftBuildService.java
@@ -20,6 +20,7 @@ import java.io.FileWriter;
 import java.io.IOException;
 import java.io.PrintWriter;
 import java.io.UnsupportedEncodingException;
+import java.util.Collections;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
@@ -30,6 +31,7 @@ import java.util.concurrent.atomic.AtomicReference;
 
 import com.google.gson.JsonObject;
 import com.google.gson.JsonPrimitive;
+import io.fabric8.kubernetes.api.model.EnvVar;
 import io.fabric8.kubernetes.api.model.KubernetesList;
 import io.fabric8.kubernetes.api.model.KubernetesListBuilder;
 import io.fabric8.kubernetes.api.model.LocalObjectReferenceBuilder;
@@ -305,6 +307,7 @@ public class OpenshiftBuildService implements BuildService {
                     .withNamespace(StringUtils.isEmpty(fromNamespace) ? null : fromNamespace)
                     .endFrom()
                     .withNoCache(checkForNocache(imageConfig))
+                    .withEnv(checkForEnv(imageConfig))
                     .endDockerStrategy().build();
             
             if (openshiftPullSecret != null) {
@@ -359,6 +362,15 @@ public class OpenshiftBuildService implements BuildService {
             BuildImageConfiguration buildConfig = imageConfig.getBuildConfiguration();
             return buildConfig.getNoCache();
         }
+    }
+
+    private List<EnvVar> checkForEnv(ImageConfiguration imageConfiguration) {
+        BuildImageConfiguration buildImageConfiguration = imageConfiguration.getBuildConfiguration();
+        if (buildImageConfiguration.getArgs() != null) {
+            return KubernetesResourceUtil.convertToEnvVarList(buildImageConfiguration.getArgs());
+        }
+
+        return Collections.emptyList();
     }
 
     private Boolean checkOrCreatePullSecret(BuildServiceConfig config, OpenShiftClient client, KubernetesListBuilder builder, String pullSecretName, ImageConfiguration imageConfig)

